ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/internal/Scalars.java

```diff
@@ -55,8 +55,13 @@ public Address parseLiteral(final Object input) throws CoercingParseLiteralExcep
           if (!(input instanceof StringValue)) {
             throw new CoercingParseLiteralException("Value is not any Address : '" + input + "'");
           }
+          String inputValue = ((StringValue) input).getValue();
+          if (!inputValue.startsWith("0x")) {
+            throw new CoercingParseLiteralException(
+                "Address value '" + inputValue + "' is not prefixed with 0x");
+          }
```

Comment:
Just for the sake of keeping things logically co-located, I'd like to see this functionality in `Quantity.java` which has a lot of utility methods related to this. Maybe a `static` method like `Quantity.isValid(String string)`?

Reply:
@RatanRSur thanks for the suggestion, I moved this check to `Qunatitiy.java` but as I read jdoc of class (_...Quantity fields are represented as minimal length hex strings with no zero-padding..._) I think this `Quantity.isValid(String string)` function should do additional checks

- is provided `string` valid hex string (_just check that characters are valid_)
- that provided hex string is not padded with `0`

or I could rename function to something that reflects what exactly is tested.

What do you think ?